### PR TITLE
CASSCF: semicanonicalize docc and virt blocks of Favg.

### PR DESCRIPTION
**Pull request by fevangelista**
## Description
Fixes an issue related to the definition of natural orbitals in CASSCF computations (#740).

- [x] Ready to go

### CODE REVIEW

File: psi4/src/psi4/detci/opdm.cc [hunk 1]
```diff
@@ -575,17 +576,61 @@ void CIWavefunction::opdm_block(struct stringwr **alplist, struct stringwr **bet
...
-  // get a copy of the active orbitals
+  // get a copy of the active orbitals and rotate them
   SharedMatrix Cactv = get_orbitals("ACT");
   SharedMatrix Cnat = Matrix::doublet(Cactv, NO_vecs);
 
   // set the active block of Ca_
   set_orbitals("ACT",Cnat);
+
+  // semicanonicalize the DOCC and VIR blocks only for CASSCF
+  if (options_.get_str("WFN") == "CASSCF"){
```
Comment (dgasmith) [inline psi4/src/psi4/detci/opdm.cc]: On second thought I do try to protect the CASSCF wavefunction can you add a `if(!somcscf_) throw ...`?

Comment (fevangelista) [inline psi4/src/psi4/detci/opdm.cc]: Ok, let me give it a try.

**Comment by dgasmith** [comment]
@fevangelista Double checking, but this is ready to go correct?

**Comment by fevangelista** [comment]
Yes, this is ready to go.

**Pull request merged**